src/reduxTypes.js

```diff
@@ -106,20 +106,24 @@ export type FetchingState = {|
  *
  * Unlike almost all other subtrees of our state, this one can be
  * incomplete, always in exactly the same way that `MessagesState` is.
+ *
+ * TODO: We currently represent "unknown message" and "message without flag"
+ * the same way, by just omitting the message ID. We may want to
+ * differentiate these.
```

Comment:
Hmm, interesting.

I think we probably don't want to do that here, because the thing is that most of these flags are very sparse: the number of messages you have starred, or where you're @-mentioned, is typically a very small fraction of the total number of messages you have. So explicitly indicating "not starred", "not @-mentioned", etc., for all those messages could be a significant increase in memory and storage use when you have a lot of messages in the state.

(That also suggests that we should be storing the *unread*, rather than the read, messages here. But even as is, we're getting the benefit of this optimization for all the other flags. And wow, there are a lot of flags there -- I have no idea what some of these mean.)

I think probably the right way to distinguish "unknown message" from "message without flag" is going to be to look up the message ID in `state.messages`.